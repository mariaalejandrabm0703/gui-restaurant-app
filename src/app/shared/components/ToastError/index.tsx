import * as React from 'react';
import { ToastContainer } from 'react-toastify';

/**
 * Componente encargado de renderizar mensaje de error
 */
const ToastError = () => {
  return (
    <div data-testid="toast-error">
      <ToastContainer
        position="top-right"
        autoClose={5000}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
      />
    </div>
  );
};
export default ToastError;
