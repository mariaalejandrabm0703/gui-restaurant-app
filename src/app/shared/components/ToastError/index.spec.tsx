import '@testing-library/jest-dom';

import { cleanup } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import ToastError from './index';

afterEach(cleanup);

it('rederizar el componente ToastError sin errores', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ToastError />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Compara snapshot del componente Loading en modo lista', () => {
  const element = renderer
    .create(
      <BrowserRouter>
        <ToastError />
      </BrowserRouter>
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});
