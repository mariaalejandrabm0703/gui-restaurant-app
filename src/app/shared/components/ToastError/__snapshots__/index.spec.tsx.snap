// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compara snapshot del componente Loading en modo lista 1`] = `
<div
  data-testid="toast-error"
>
  <div
    className="Toastify"
  />
</div>
`;
